src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -31,11 +31,11 @@ public CertificateCallbackMapper(Func<HttpRequestMessage, X509Certificate2?, X50
             }
         }
 
-        public static async ValueTask<Stream> ConnectAsync(SocketsConnectionFactory factory, DnsEndPoint endPoint, CancellationToken cancellationToken)
+        public static async ValueTask<Stream> ConnectAsync(Func<DnsEndPoint, HttpRequestMessage, CancellationToken, ValueTask<Stream>> callback, DnsEndPoint endPoint, HttpRequestMessage requestMessage, CancellationToken cancellationToken)
```

Comment:
Why do we still need SocketsConnectionFactory at all?

Reply:
We don't, really. I originally kept it around to try to minimize the amount of code change. But I'm not sure it was worth it.

Reply:
This was removed.